File: src/main/java/no/novari/flyt/egrunnerverv/gateway/instance/model/EgrunnervervJournalpostInstance.java
package no.novari.flyt.egrunnerverv.gateway.instance.model;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.extern.jackson.Jacksonized;

@Getter
@Jacksonized
@EqualsAndHashCode
@Builder
public class EgrunnervervJournalpostInstance {
    @NotNull
    @Valid
    private final EgrunnervervJournalpostInstanceBody egrunnervervJournalpostInstanceBody;
    @NotNull
    private final String saksnummer;
}

File: src/main/kotlin/no/novari/flyt/egrunnerverv/gateway/instance/model/EgrunnervervJournalpostInstance.kt
package no.novari.flyt.egrunnerverv.gateway.instance.model

import jakarta.validation.Valid

data class EgrunnervervJournalpostInstance(
    @field:Valid
    val egrunnervervJournalpostInstanceBody: EgrunnervervJournalpostInstanceBody,
    val saksnummer: String,
)
